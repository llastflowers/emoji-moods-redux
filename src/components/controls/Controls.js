import React from 'react';
import styles from './Controls.css';
import { getCoffees, getSnacks, getNaps, getStudies } from '../../selectors/moodSelectors';
import { useSelector, useDispatch } from 'react-redux';
import { drinkCoffee, eatSnack, takeNap, study } from '../../actions/emojiActions';

const Controls = () => {
  const dispatch = useDispatch();
  const coffees = useSelector(getCoffees);
  const snacks = useSelector(getSnacks);
  const naps = useSelector(getNaps);
  const studies = useSelector(getStudies);
  return (
    <section className={styles.Controls}>
      <button onClick={() => dispatch(drinkCoffee())}>coffee - {coffees}</button>
      <button onClick={() => dispatch(eatSnack())}>snacks - {snacks}</button>
      <button onClick={() => dispatch(takeNap())}>naps - {naps}</button>
      <button onClick={() => dispatch(study())}>studies - {studies}</button>
    </section>
  );
};

export default Controls;
