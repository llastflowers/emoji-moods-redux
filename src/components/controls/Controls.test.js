import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Controls component', () => {
  let wrapper;
  let handleSelection;
  beforeEach(() => {
    const actions = [
      { name: 'DO_STUFF', text: 'stuff' },
      { name: 'other' }
    ];
    handleSelection = jest.fn();

    const wrapper = shallow(<Provider store={store}><Controls
      actions={actions}
      handleSelection={handleSelection} /></Provider>).dive;
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Controls', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
  
