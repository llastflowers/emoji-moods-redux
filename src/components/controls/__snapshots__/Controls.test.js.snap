// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `[Function]`;

exports[`Controls component renders Controls 2`] = `undefined`;
